src/test/java/duke/ParserTest.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.tasklist.TaskList;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ParserTest {
+
+    private TaskList taskList;
+    private Parser parser;
+
+    @BeforeEach
+    public void init() {
+        this.taskList = new TaskList();
+        this.parser = new Parser(taskList);
+    }
+
+    @Test
+    public void parseTest() {
+        try {
+            Command command = parser.parse("Hi");
+            assertTrue(command instanceof StartCommand);
+
+            command = parser.parse("list");
+            assertTrue(command instanceof ListCommand);
+
+            command = parser.parse("todo read book");
+            assertTrue(command instanceof ToDoCommand);
+
+            command = parser.parse("deadline borrow book /by 2022-02-02 1900");
+            assertTrue(command instanceof DeadlineCommand);
+
+            command = parser.parse("event return book /from 2022-02-02 1800 /to 2023-01-02 1722");
+            assertTrue(command instanceof EventCommand);
+
+            command = parser.parse("bye");
+            assertTrue(command instanceof ExitCommand);
+        } catch (Exception e) {
+            fail("Unexpected DukeException: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void invalidCommandTest() {
```

Comment:
Should this be renamed to `testParse_invalidInputs_exceptionThrown`? I notice that this issue can be seen elsewhere in the codebase.